### PR TITLE
messaging: collapse stale draft notifications

### PR DESCRIPTION
**Pull request by elie222**
# User description
Updates messaging draft cleanup when a reply is sent outside the channel.

Slack draft cards now collapse to a handled state instead of keeping the full interactive history. Reply tracking triggers the cleanup while preserving the existing handled state for replies sent from Slack. Adds targeted tests for notification cleanup and outbound draft tracking.

---

# Generated description

Below is a concise technical summary of the changes proposed in this PR:
Collapse stale messaging draft notifications by updating <code>replaceMessagingDraftNotificationsWithHandledOnWebState</code> so provider cards expire and show a handled state once a reply is sent outside the channel. Trigger the same cleanup from <code>trackSentDraftStatus</code> when the reply tracker records a web send, keeping notification state in sync across Slack, Teams, and Telegram.
<table><tr><th>Topic</th><th>Details</th><tr><td><a href=https://baz.co/changes/elie222/inbox-zero/2278?tool=ast&topic=Draft+notification+cleanup>Draft notification cleanup</a>
        </td><td>Update <code>replaceMessagingDraftNotificationsWithHandledOnWebState</code> to expire draft actions, build a terminal card, and edit Slack/Teams/Telegram notifications after web replies while covering the new behavior with targeted tests for each provider and failure scenario.<details><summary>Modified files (2)</summary><ul><li>apps/web/utils/messaging/rule-notifications.test.ts</li>
<li>apps/web/utils/messaging/rule-notifications.ts</li></ul></details><details><summary>Latest Contributors(1)</summary><table><tr><th>User</th><th>Commit</th><th>Date</th></tr><tr><td>elie222</td><td>rules: streamline acti...</td><td>April 16, 2026</td></tr></table></details></td></tr>
<tr><td><a href=https://baz.co/changes/elie222/inbox-zero/2278?tool=ast&topic=Reply+tracking+hook>Reply tracking hook</a>
        </td><td>Invoke <code>replaceMessagingDraftNotificationsWithHandledOnWebState</code> from <code>trackSentDraftStatus</code> and its tests so reply tracking now collapses stale draft notifications whenever a web reply is recorded.<details><summary>Modified files (2)</summary><ul><li>apps/web/utils/reply-tracker/draft-tracking.test.ts</li>
<li>apps/web/utils/reply-tracker/draft-tracking.ts</li></ul></details><details><summary>Latest Contributors(2)</summary><table><tr><th>User</th><th>Commit</th><th>Date</th></tr><tr><td>elie222</td><td>fix: handle quoted hid...</td><td>April 04, 2026</td></tr>
<tr><td>mojkakec12345@gmail.com</td><td>fix webhook email proc...</td><td>July 04, 2025</td></tr></table></details></td></tr></table>
This pull request is reviewed by Baz. Review like a pro on <a href=https://baz.co/changes/elie222/inbox-zero/2278?tool=ast>(Baz)</a>.

### CODE REVIEW

File: apps/web/utils/messaging/rule-notifications.ts [hunk 1]
```diff
@@ -127,6 +127,38 @@ export async function sendMessagingRuleNotification({
...
+    },
+  });
+
+  await Promise.all(
+    notificationActions.map(({ id }) =>
+      replaceSlackDraftNotificationWithHandledOnWebState({
+        executedActionId: id,
+        logger,
+      }),
+    ),
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/utils/messaging/rule-notifications.ts]: Applied. The fan-out now uses `Promise.allSettled()` and logs per-item failures so one thrown cleanup task does not make the exported helper reject. I also added a regression test covering a failing lookup alongside a successful collapse path.

File: apps/web/utils/messaging/rule-notifications.ts [hunk 2]
```diff
@@ -1591,6 +1632,183 @@ async function postSlackCard({
...
+      !context.messagingChannel.accessToken)
+  ) {
+    return;
+  }
+
+  if (
+    context.messagingMessageStatus !== MessagingMessageStatus.SENT &&
+    context.messagingMessageStatus !== MessagingMessageStatus.DRAFT_EDITED
+  ) {
+    return;
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/utils/messaging/rule-notifications.ts:1695]: Valid. The helper now treats legacy `messagingMessageStatus: null` rows as active for the local expiration step by including `null` alongside `SENT` and `DRAFT_EDITED` in the `updateMany` filter. I also added test coverage for the legacy-null case.

File: apps/web/utils/messaging/rule-notifications.ts [hunk 3]
```diff
@@ -1591,6 +1632,183 @@ async function postSlackCard({
...
+}) {
+  const context = await getNotificationContext(executedActionId);
+
+  if (
+    !context?.messagingMessageId ||
+    !context.messagingChannel?.isConnected ||
+    (context.messagingChannel.provider === MessagingProvider.SLACK &&
+      !context.messagingChannel.accessToken)
+  ) {
+    return;
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/utils/messaging/rule-notifications.ts]: Valid. I moved the local `EXPIRED` transition ahead of the remote-edit capability checks, so the row now expires even if the channel is disconnected or Slack is missing an access token. Remote edit/collapse still short-circuits afterward with a warning when it cannot run. Covered with a regression test.
